subsys/bluetooth/mesh/Kconfig

```diff
@@ -434,145 +337,60 @@ config BT_MESH_MSG_CACHE_SIZE
 
-config BT_MESH_ADV_BUF_COUNT
-	int "Number of advertising buffers for local messages"
-	default 6
-	range 1 256
-	help
-	  Number of advertising buffers available for sending local messages.
-	  This should be chosen based on the number of local messages that the node
-	  can send simultaneously.
-
-choice BT_MESH_ADV
-	prompt "Advertiser mode"
-	default BT_MESH_ADV_EXT if BT_EXT_ADV
-	default BT_MESH_ADV_LEGACY
-
-config BT_MESH_ADV_LEGACY
-	bool "Legacy advertising"
-	help
-	  Use legacy advertising commands for mesh sending. Legacy
-	  advertising is significantly slower than the extended advertising, but
-	  is supported by all controllers.
-
-	  WARNING: The legacy advertiser can occasionally do more message
-	  retransmissions than requested because of limitations of HCI
-	  interface API.
-
-config BT_MESH_ADV_EXT
-	bool "Extended advertising"
-	depends on BT_CTLR_ADV_EXT || !BT_CTLR
-	depends on BT_EXT_ADV
-	help
-	  Use extended advertising commands for operating the advertiser.
-	  Extended advertising is faster and uses less memory than legacy
-	  advertising, but isn't supported by all controllers.
-
-endchoice
-
-if BT_MESH_ADV_EXT
-
-config BT_MESH_RELAY_ADV_SETS
-	int "Maximum of simultaneous relay message support"
-	default 0
-	range 0 BT_EXT_ADV_MAX_ADV_SET
-	depends on BT_MESH_RELAY
-	help
-	  Maximum of simultaneous relay message support. Requires controller support
-	  multiple advertising sets.
-
-config BT_MESH_ADV_EXT_RELAY_USING_MAIN_ADV_SET
-	bool "Use the main advertising set to relay messages"
-	depends on BT_MESH_RELAY_ADV_SETS > 0
-	help
-	  When this option is enabled, there is a message that needs to be
-	  relayed, all relay advertising sets defined by
-	  CONFIG_BT_MESH_RELAY_ADV_SETS are busy with relaying messages
-	  and the main advertising set is not busy with sending local
-	  messages, the stack will use the main advertising set to relay
-	  the message. This maximizes the utilization efficiency of
-	  advertising sets, which is helpful for the sending of dense
-	  relays. With CONFIG_BT_MESH_RELAY_RETRANSMIT_COUNT value
-	  greater than zero, this can noticeably delay transmission of
-	  local messages. When Friend feature is enabled and the node is
-	  in a friendship, this option can delay transmission of local
-	  messages thus requiring bigger CONFIG_BT_MESH_FRIEND_RECV_WIN
-	  value. This in turn will result in increase of the power
-	  consumption of the Low Power node.
-
-config BT_MESH_ADV_EXT_GATT_SEPARATE
-	bool "Use a separate extended advertising set for GATT Server Advertising"
-	depends on BT_MESH_GATT_SERVER
-	help
-	  Use a separate extended advertising set for GATT Server Advertising,
-	  otherwise a shared advertising set will be used.
-
-config BT_MESH_ADV_EXT_FRIEND_SEPARATE
-	bool "Use a separate extended advertising set for Friend advertising"
-	depends on BT_MESH_FRIEND
+menuconfig BT_MESH_RELAY
+	bool "Relay support"
```

Comment:
Is not this option basically the same as `BT_MESH_RELAY_ENABLED`? Does it make sence to activate relay support if `BT_MESH_RELAY_ENABLED` is set to `n`? I guess this option could replace it, unless it is a deprecation thing.

Reply:
Same deal for proxy and for friend as well, maybe there is more.

Reply:
Ahh, ok. I see the difference now. `Can be changed through runtime configuration`: So this decides if the device boots with the feature enabled. Then I guess it is fine, but it is was a bit tricky for me to catch this from the text. Don't know if it is possible to clarify this further in the text of these options.

Reply:
Tried to clarify this a bit. Please check if it is better now.